galsim/scene.py

```diff
@@ -86,46 +92,58 @@ class COSMOSCatalog(object):
-                        and hlr > 1 arcsec, probably indicates poor sky subtraction. [default: True]
-    @param min_hlr      Exclude galaxies whose fitted half-light-radius is smaller than this value
-                        (in arcsec).  [default: 0, meaning no limit]
-    @param max_hlr      Exclude galaxies whose fitted half-light-radius is larger than this value
-                        (in arcsec).  [default: 0, meaning no limit]
-    @param min_flux     Exclude galaxies whose fitted flux is smaller than this value.  
-                        [default: 0, meaning no limit]
-    @param max_flux     Exclude galaxies whose fitted flux is larger than this value.  
-                        [default: 0, meaning no limit]
+    @param file_name        The file containing the catalog. [default: None, which will look for the
+                            F814W<25.2 COSMOS catalog in $PREFIX/share/galsim.  It will raise an
+                            exception if the catalog is not there telling you to run
+                            galsim_download_cosmos.]
+    @param sample           A keyword argument that can be used to specify the sample to use, i.e.,
+                            "23.5" or "25.2".  At most one of `file_name` and `sample` should be
+                            specified.
+                            [default: None, which results in the same default as `file_name=None`.]
+    @param image_dir        Keyword that is only used for real galaxies, not parametric ones, to
+                            specify the directory of the image files.
+                            If a string containing no `/`, it is the relative path from the location
+                            of the catalog file to the directory containing the galaxy/PDF images.
+                            If a path (a string containing `/`), it is the full path to the
+                            directory containing the galaxy/PDF images. [default: None]
+    @param dir              The directory of catalog file. [default: None]
+    @param preload          Keyword that is only used for real galaxies, not parametric ones, to
+                            choose whether to preload the header information.  If `preload=True`,
+                            the bulk of the I/O time is in the constructor.  If `preload=False`,
+                            there is approximately the same total I/O time (assuming you eventually
+                            use most of the image files referenced in the catalog), but it is spread
+                            over the various calls to getGal() and getPSF().  [default: False]
+    @param noise_dir        Keyword that is only used for real galaxies, not parametric ones.
+                            The directory of the noise files if different from the directory of the 
+                            image files.  [default: image_dir]
+    @param use_real         Enable the use of realistic galaxies?  [default: True]
```

Comment:
Well - in the very original version of the code I think there may have been more benefit because it was structured differently.  Now, there is a quite small benefit in terms of RAM and speed (don't have to read in the real catalog) and a possible logistical benefit, like if you never want to use realistic galaxies and deleted the real catalog from your system, you could still use COSMOSCatalog with ```use_real=False``` and not have it complain that it couldn't find the catalog.  I just added a note on this.